### ISSUE TITLE
 ERROR: Failed deflating readstring

### ISSUE THREAD

**Issue by Enorya** [biocore/sortmerna#318]
Hello,

I am trying to use sortmerna on a set of reads but I got the following error from the Readfeed init step: 
```
ERROR: Failed deflating readstring: @SRR12660679.1286586.2 1286586 length=100
GAGGGATTTCCTGCAAAACTGGAAGTTAAAGCTGTTGGGCATCCCAAACCGACTCTGAAATGGACGCATAATGGGAAAGAAATTGTGCCTGATGGGGAAA
+
AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
 Output file idx: 1 zlib status: -1
```
I also have another error afterwards during the alignment. It seems that one processor didn't work because I have 6 processors at the step `[align2:70]` and then one dissapear at the following step `[next:455] EOF FWD reached. Total reads: 1` and I got this message at the step `[align2:133]`:
```
[next:462] ERROR: reading from FWD file. Exiting...
```
Can you tell me what those errors mean?
How can I correct them?

Thank you in advance for your help,
Enora Geslain

**Comment by biocodz** [comment]
Hi. 
Would it be possible, please, to provide the [trace](https://github.com/biocore/sortmerna#execution-trace)

**Comment by Enorya** [comment]
Hi,

I thought I kept the trace but I probably deleted it last time...
I'm going to redo the analysis and I will send you the trace as soon as I have it.
For the `sortmerna  --version` this is the output:
```
[process:1376] === Options processing starts ... ===
Found value: sortmerna
Found flag: --version
[process:1460] Processing option: version with value:

SortMeRNA version 4.3.4
Build Date: Jul 21 2021
sortmerna_build_git_sha:@90cdf6ce459e9477c5d0de9a5daab025f49f3cc6@
sortmerna_build_git_date:@2021/07/21 11:56:36@
```

**Comment by biocodz** [comment]
Hi Enora,
I looked at the trace you've sent me.
The program fails to compress the reads into splits although it has not problem reading the input files. I didn't see this problem before, so I would need to work with the actual files to be able to figure out what is going on.
The things I'd try prior getting into the debugger
- rm /user/leuven/340/vsc34088/sortmerna/run/readb/* #                     clean-up the split files and re-ran the program
- test the program on a subset of the records say 10K or 100K from each file. 
- try decompressing the reads and running the program on the flat files.

**Comment by Enorya** [comment]
Hi,

I tried the debugging you suggested:

- removing the readb/ folder content seems to have no effect on the result, I still have the same errors in the trace as you can see in the following document: [SRR12660679_sortmerna.log](https://github.com/biocore/sortmerna/files/8283645/SRR12660679_sortmerna.log)
- I tried to run the analysis on a decompressed subset of 10k lines from my fastq files and this time it worked perfectly as you can see in the following trace: [SRR12660679_sortmerna_subset.log](https://github.com/biocore/sortmerna/files/8283680/SRR12660679_sortmerna_subset.log)
- I'm triyng right now to run the analysis on the complete decompressed fastq files but it takes some time so I don't have the results for the moment.

I sent you the fastq files I'm using with WeTransfer so you can have a look at it yourself. You should receive them on your email address. Hopefully you will find what is wrong with my files.
To run sortmerna I'm using the following script (my goal is to retrieve all non rRNA reads and for the database I'm using a concatenation of all rRNA databases available in your data directory):
```
sortmerna --fastx --other $OUTDIR/SRR12660679_sortmerna --out2 --idx-dir $OUTDIR/rRNA_databases/idx \
--paired_in --reads SRR12660679_R1.fastq.gz --reads SRR12660679_R2.fastq.gz --ref $DB --threads 6 \
--kvdb SRR12660679_kvdb --task 4 |& tee $OUTDIR/SRR12660679_sortmerna.log
```

**Comment by Enorya** [comment]
Hi, 

- the trial with decompressed files is finished, it seems that everything worked correctly but I'm surprised by the number of non rRNA retrieved: less than 1 million for a total of 76 million reads. Because of this, I'm not sure it ran correctly. Can you tell me if you see something that could explain this low number in the trace of the run: [SRR12660679_sortmerna_decompressed.log](https://github.com/biocore/sortmerna/files/8303060/SRR12660679_sortmerna_decompressed.log)

Regarding the error, it seems that it disappears when I use decompressed fastq files. Do you have any idea why this problem could occur?

**Comment by biocodz** [comment]
@Enorya 
I received your files and will let you know the results soon

**Comment by biocodz** [comment]
I was not able to reproduce the problem using 8 threads, 2 references, and a slightly simplified output
```
sortmerna -ref /media/sf_a01_code/sortmerna/data/silva-bac-16s-database-id85.fasta -ref /media/sf_a01_code/sortmerna/data/silva-arc-16s-database-id95.fasta -reads /media/sf_a01_data/sortmerna/bugs/iss_318_gzip_fails/SRR12660679_R1.fastq.gz -reads /media/sf_a01_data/sortmerna/bugs/iss_318_gzip_fails/SRR12660679_R2.fastq.gz -fastx -other -blast 1 -v -workdir /home/biocodz/sortmerna/run -threads 8 -dbg-level 2 -task 4
```
The whole process took ~40min on my Dell XPS15 i9 laptop with 32G memory with VBox VM Ubuntu 16.04 on Win11 host , out of which the splitting took ~26min, which is OK for 40M reads.

The split was ok
```
...
[split:717] Done splitting. Reads count: 38209778 Runtime sec: 1489.79
[init:135] Readfeed init done in sec [1579.48]
...
```
The split produced 16 files as expected
```
ls -l /home/biocodz/sortmerna/run/readb/
total 1964240
-rw-rw-r-- 1 biocodz biocodz 123448209 Mar 23 07:35 fwd_0.fq.gz
-rw-rw-r-- 1 biocodz biocodz 122573970 Mar 23 07:38 fwd_1.fq.gz
-rw-rw-r-- 1 biocodz biocodz 123065719 Mar 23 07:41 fwd_2.fq.gz
-rw-rw-r-- 1 biocodz biocodz 122373248 Mar 23 07:44 fwd_3.fq.gz
-rw-rw-r-- 1 biocodz biocodz 125493959 Mar 23 07:47 fwd_4.fq.gz
-rw-rw-r-- 1 biocodz biocodz 124508816 Mar 23 07:50 fwd_5.fq.gz
-rw-rw-r-- 1 biocodz biocodz 122980928 Mar 23 07:54 fwd_6.fq.gz
-rw-rw-r-- 1 biocodz biocodz 123554711 Mar 23 07:57 fwd_7.fq.gz
-rw-rw-r-- 1 biocodz biocodz      1609 Mar 23 07:57 readfeed
-rw-rw-r-- 1 biocodz biocodz 125664611 Mar 23 07:35 rev_0.fq.gz
-rw-rw-r-- 1 biocodz biocodz 125978381 Mar 23 07:38 rev_1.fq.gz
-rw-rw-r-- 1 biocodz biocodz 126400130 Mar 23 07:41 rev_2.fq.gz
-rw-rw-r-- 1 biocodz biocodz 126075930 Mar 23 07:44 rev_3.fq.gz
-rw-rw-r-- 1 biocodz biocodz 134199936 Mar 23 07:47 rev_4.fq.gz
-rw-rw-r-- 1 biocodz biocodz 130802756 Mar 23 07:50 rev_5.fq.gz
-rw-rw-r-- 1 biocodz biocodz 126486503 Mar 23 07:54 rev_6.fq.gz
-rw-rw-r-- 1 biocodz biocodz 127671450 Mar 23 07:57 rev_7.fq.gz
```
The produced output files are all OK
```
ls -lrth /home/biocodz/sortmerna/run/out
total 1.9G
-rw-rw-r-- 1 biocodz biocodz 1.9K Mar 23 08:05 aligned.log
-rw-rw-r-- 1 biocodz biocodz 229K Mar 23 08:10 aligned.fq.gz
-rw-rw-r-- 1 biocodz biocodz 1.9G Mar 23 08:10 other.fq.gz
-rw-rw-r-- 1 biocodz biocodz  94K Mar 23 08:10 aligned.blast.gz
```
Very few RNA reads identified
```
cat /home/biocodz/sortmerna/run/out/aligned.log
...
    Reads file: /media/sf_a01_data/sortmerna/bugs/iss_318_gzip_fails/SRR12660679_R1.fastq.gz
    Reads file: /media/sf_a01_data/sortmerna/bugs/iss_318_gzip_fails/SRR12660679_R2.fastq.gz
    Total reads = 38209778

 Results:
    Total reads passing E-value threshold = 7058 (0.02)
    Total reads failing E-value threshold = 38202720 (99.98)
    ...

 Coverage by database:
    /media/sf_a01_code/sortmerna/data/silva-bac-16s-database-id85.fasta         0.01
    /media/sf_a01_code/sortmerna/data/silva-arc-16s-database-id95.fasta         0.00
```

You can verify the correctness of the results using online tools like BLAST. Just pick-up a few random reads from 'other.fq.gz'.

There is also a newer RNA DB available [here](https://github.com/biocore/sortmerna/releases/download/v4.3.4/database.tar.gz) that could potentially give a better coverage

**Comment by Enorya** [comment]
Hi, 

I finally realised what was the problem. I'm working on a cluster and the default place used by Sortmerna to store the different outputs was my home directory and there wasn't enough space on it. So, to solve the problem I used the parameter `--workdir` to change the default path for the different outputs. It's now working perfectly and using the newer RNA DB available sortmerna found actually more rRNA in my reads (1.97% in total).

Thank you for your help!
